src/main.rs

```diff
@@ -152,8 +158,22 @@ impl Leroy {
     fn handle_line(&mut self, line: Vec<u8>) {
         self.line_count += 1;
 
-        let ban_log_count: u32 = *self.ban_log_count_cache.get(&line).unwrap_or(&0) + 1;
-        if ban_log_count >= self.args.bl_threshold {
+        if !self.ip_rate_limiters.contains_key(&line) {
+            self.ip_rate_limiters.insert(
+                line.clone(),
+                RateLimiter::direct(
+                    Quota::with_period(self.args.bl_period)
+                        .expect("Rate limits MUST Be non-zero")
+                        .allow_burst(self.args.bl_threshold),
+                ),
+            );
+        }
+
+        let rate_limiter = self
+            .ip_rate_limiters
+            .get(&line)
+            .expect("This should never fail :kappa:");
```

Comment:
This can fail if the value has expired in the short time window between `contains_key()` and `get()`.